backend/app/src/main/java/com/ugent/pidgeon/controllers/TestController.java

```diff
@@ -53,83 +53,96 @@ public class TestController {
+
+
+        if (!updateCheckResult.getStatus().equals(HttpStatus.OK)) {
+            return ResponseEntity.status(updateCheckResult.getStatus()).body(updateCheckResult.getMessage());
         }
-        TestEntity testEntity = checkResult.getData().getFirst();
-        ProjectEntity projectEntity = checkResult.getData().getSecond();
-
-        try {
-
-            // Save the files on server
-            long dockertestFileEntityId;
-            long structuretestFileEntityId;
-            if (dockerTest != null) {
-                Path dockerTestPath = Filehandler.saveTest(dockerTest, projectId);
-                FileEntity dockertestFileEntity = fileUtil.saveFileEntity(dockerTestPath, projectId, user.getId());
-                dockertestFileEntityId = dockertestFileEntity.getId();
-            } else {
-                dockertestFileEntityId = testEntity.getDockerTestId();
-            }
 
-            if (structureTest != null) {
-                Path structureTestPath = Filehandler.saveTest(structureTest, projectId);
-                FileEntity structuretestFileEntity = fileUtil.saveFileEntity(structureTestPath, projectId, user.getId());
-                structuretestFileEntityId = structuretestFileEntity.getId();
-            } else {
-                structuretestFileEntityId = testEntity.getStructureTestId();
+        TestEntity testEntity = updateCheckResult.getData().getFirst();
+        ProjectEntity projectEntity = updateCheckResult.getData().getSecond();
+
+        // delete test entry
+        if(httpMethod.equals(HttpMethod.DELETE)){
+            // first check if docker image is not used anywhere else
+            if(!testRepository.imageIsUsed(dockerImage)){
+                // image is no longer required for any tests
+                DockerSubmissionTestModel.removeDockerImage(dockerImage);
             }
 
-            // Create/update test entity
-            TestEntity test = new TestEntity(dockerImage, dockertestFileEntityId, structuretestFileEntityId);
-            test = testRepository.save(test);
-            projectEntity.setTestId(test.getId());
+            // delete test
+            testRepository.deleteById(testEntity.getId());
+            projectEntity.setTestId(null);
             projectRepository.save(projectEntity);
-            return ResponseEntity.ok(entityToJsonConverter.testEntityToTestJson(test, projectId));
-        } catch (IOException e) {
-            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body("Error while saving files: " + e.getMessage());
+            return ResponseEntity.ok().build();
         }
+
+        // Docker test
+        if(!(dockerImage == null && dockerScript == null && dockerTemplate == null)){
+
+            // update/install image if possible.
+            DockerSubmissionTestModel.installImage(dockerImage);
```

Comment:
Is dit is wat mogelijks een tijdje kan duren? Het lijkt me niet handig dat er gewacht moet worden op een response indien dit installeren even duurt

Reply:
Inderdaad dit kan wel een tijdje duren. Ik zal dit in een apparte thread doen